crates/orderbook/src/database/total_surplus.rs

```diff
@@ -0,0 +1,75 @@
+use {
+    anyhow::Result,
+    database::{byte_array::ByteArray, Address},
+    primitive_types::{H160, U256},
+    sqlx::PgConnection,
+};
+
+/// Computes a user's total surplus received (price improvement over limit price
+/// and **NOT** quoted price) since march 2023.
+async fn fetch_total_surplus(ex: &mut PgConnection, user: &Address) -> Result<f64, sqlx::Error> {
+    const TOTAL_SURPLUS_QUERY: &str = r#"
+WITH regular_orders AS (
+    SELECT ARRAY_AGG(uid) AS ids FROM orders WHERE owner = $1
```

Comment:
I think the total memory used probably grows with the number of orders as well. So from that point of view having these ids in memory probably only adds a relatively small percentage to the overall memory consumption.
I'm fine with that when it reduces the runtime from ~4s to ~0.4s.

Reply:
In the original query the memory doesn't need to grow with the number of orders because the surplus of each order can be determined individually and is summed to get the total.